app/src/main/java/com/duckduckgo/app/browser/localstorage/LocalStorageSettingsJsonParser.kt

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2024 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.app.browser.localstorage
+
+import com.duckduckgo.common.utils.DispatcherProvider
+import com.duckduckgo.di.scopes.AppScope
+import com.squareup.anvil.annotations.ContributesBinding
+import com.squareup.moshi.JsonAdapter
+import com.squareup.moshi.Moshi
+import com.squareup.moshi.kotlin.reflect.KotlinJsonAdapterFactory
+import javax.inject.Inject
+import kotlinx.coroutines.withContext
+
+data class Domains(val list: List<String> = emptyList())
+data class MatchingRegex(val list: List<String> = emptyList())
+
+data class LocalStorageSettings(
+    val domains: Domains = Domains(),
+    val matchingRegex: MatchingRegex = MatchingRegex(),
+)
+
+interface LocalStorageSettingsJsonParser {
+    suspend fun parseJson(json: String?): LocalStorageSettings
+}
+
+@ContributesBinding(AppScope::class)
+class LocalStorageSettingsJsonParserImpl @Inject constructor(
+    private val dispatcherProvider: DispatcherProvider,
+) : LocalStorageSettingsJsonParser {
+
+    private val jsonAdapter by lazy { buildJsonAdapter() }
+
+    private fun buildJsonAdapter(): JsonAdapter<SettingsJson> {
+        val moshi = Moshi.Builder().add(KotlinJsonAdapterFactory()).build()
```

Comment:
Note: I wonder if we should directly add the `KotlinJsonAdapterFactory` to the moshi instance that it's in DI. 
I see we're creating new instances of moshi everywhere just because the DI instance doens't have that adapter

Also no need to do it in this PR